packages/nimble-components/src/wafer-map/modules/experimental/hover-handler.ts

```diff
@@ -0,0 +1,102 @@
+import type { WaferMap } from '../..';
+import {
+    PointCoordinates,
+    WaferMapOriginLocation,
+    WaferRequiredTypeMap
+} from '../../types';
+
+/**
+ * HoverHandler deals with user interactions and events like hovering
+ */
+export class HoverHandler<T extends WaferRequiredTypeMap> {
+    public constructor(private readonly wafermap: WaferMap<T>) {}
```

Comment:
We don't need to propagate the generic  into all of these classes. Most if not all of these classes at most care about the required fields which are already satisfied by the default type you specified here:
```ts
export class WaferMap<
    T extends WaferRequiredFields = WaferRequiredFields
> extends FoundationElement {
```

Almost none of these classes actually care if there are extra fields specified by the user, they only operate on the required fields. So we don't need to propagate the generic everywhere, only where it's needed (when we need to know more than just the required fields)

```suggestion
import {
    PointCoordinates,
    WaferMapOriginLocation
} from '../../types';

/**
 * HoverHandler deals with user interactions and events like hovering
 */
export class HoverHandler {
    public constructor(private readonly wafermap: WaferMap) {}
```

So lets not add the generic to all these. It makes it more complex without any benefit

Reply:
I removed the generics from the modules but there are still type mismatches when initializing them

Reply:
Yea that's interesting. I sunk the changes locally and I think it has to do with Apache Arrow's slightly strange typing on table fields. I pushed a change that does a couple things:
- Made an accessor that performs the `as` cast so that they are not sprinkled in many places
- Updated mocks so that they cast to the mocked type before returning, this let us avoid lots of `as` casts in tests
- Modified the validator code to use references to the field objects instead of indexes in the arrays. This helped improve TS inferencing so many non-null assertions (`!`) could be removed and made it easier to read.